update all the static check tools to latest version

I was going through the golangci-lint new release, looks like we have some static issues its good to run all static tools with latest versions

```go
make go-lint 
cephcsi image settings: quay.io/cephcsi/cephcsi version canary
./scripts/lint-go.sh
INFO [config_reader] Used config file scripts/golangci.yml 
INFO [lintersdb] Active 46 linters: [asciicheck bodyclose deadcode depguard dogsled errcheck exhaustive exportloopref gci gocognit goconst gocritic gocyclo godot gofmt gofumpt goheader goimports golint gomnd gomodguard goprintffuncname gosec gosimple govet ineffassign interfacer lll maligned misspell nakedret nlreturn noctx nolintlint rowserrcheck scopelint sqlclosecheck staticcheck structcheck stylecheck typecheck unconvert unparam unused varcheck whitespace] 
INFO [loader] Using build tags: [nautilus]        
INFO [loader] Go packages loading at mode 575 (compiled_files|exports_file|imports|name|deps|files|types_sizes) took 1.362612936s 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 21.267923ms 
INFO [linters context/goanalysis] analyzers took 2m30.726592724s with top 10 stages: buildir: 2m1.340534731s, inspect: 3.066803154s, buildssa: 2.741827785s, the_only_name: 2.046159586s, exhaustive: 1.9981591s, gocritic: 1.819492945s, goimports: 1.801243659s, fact_deprecated: 1.495034764s, ctrlflow: 1.397161224s, gofumpt: 1.390896208s 
INFO [linters context/goanalysis] analyzers took 2.318403989s with top 10 stages: buildir: 2.070186758s, U1000: 248.217231ms 
WARN [runner/nolint] Found unknown linters in //nolint directives: gocognit:gocyclo 
INFO [runner/max_same_issues] 508/511 issues with text "return with no blank line before" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 52/55 issues with text "File is not `gofumpt`-ed" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 47/50 issues with text "File is not `gci`-ed with -local github.com/ceph/ceph-csi" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 38/41 issues with text "Top level comment should end in a period" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 3/6 issues with text "break with no blank line before" were hidden, use --max-same-issues 
INFO [runner] Issues before processing: 929, after processing: 21 
INFO [runner] Processors filtering stat (out/in): path_shortener: 21/21, autogenerated_exclude: 929/929, exclude: 756/929, nolint: 672/756, diff: 669/669, max_per_file_from_linter: 669/669, max_same_issues: 21/669, filename_unadjuster: 929/929, max_from_linter: 21/21, source_code: 21/21, severity-rules: 21/21, sort_results: 21/21, cgo: 929/929, skip_files: 929/929, uniq_by_line: 669/672, path_prefixer: 21/21, path_prettifier: 929/929, skip_dirs: 929/929, identifier_marker: 929/929, exclude-rules: 756/756 
INFO [runner] processing took 78.78974ms with stages: nolint: 36.903139ms, exclude: 19.474585ms, identifier_marker: 17.716477ms, path_prettifier: 1.88781ms, autogenerated_exclude: 1.598742ms, source_code: 394.804µs, cgo: 281.723µs, skip_dirs: 194.151µs, uniq_by_line: 130.242µs, max_same_issues: 111.776µs, filename_unadjuster: 49.069µs, max_per_file_from_linter: 35.453µs, max_from_linter: 5.517µs, path_shortener: 4.469µs, diff: 486ns, skip_files: 346ns, exclude-rules: 340ns, sort_results: 281ns, severity-rules: 196ns, path_prefixer: 134ns 
INFO [runner] linters took 43.53724278s with stages: goanalysis_metalinter: 41.594033125s, unused: 1.864306305s 
internal/util/crypto.go:20: File is not `gci`-ed with -local github.com/ceph/ceph-csi (gci)
	"context"
internal/util/crypto.go:28: File is not `gci`-ed with -local github.com/ceph/ceph-csi (gci)

	"crypto/rand"
internal/journal/omap.go:22: File is not `gci`-ed with -local github.com/ceph/ceph-csi (gci)

internal/util/cephconf.go:35:42: Top level comment should end in a period (godot)
	// CephConfigPath ceph configuration file
	                                        ^
internal/util/connection.go:39:14: Top level comment should end in a period (godot)
	// available)
	            ^
internal/util/crypto.go:36:49: Top level comment should end in a period (godot)
	// Encryption passphrase location in K8s secrets
	                                               ^
internal/util/credentials.go:115: File is not `gofumpt`-ed (gofumpt)
	var c = &Credentials{}
internal/util/topology.go:135: File is not `gofumpt`-ed (gofumpt)
	var (
		topologyPools []TopologyConstrainedPool
	)
internal/util/volid.go:93: File is not `gofumpt`-ed (gofumpt)
	return strings.Join([]string{versionEncodedHex, clusterIDLength, ci.ClusterID,
		poolIDEncodedHex, ci.ObjectUUID}, "-"), nil
internal/util/vault.go:309:16: G402: TLS MinVersion too low. (gosec)
	tlsConfig := &tls.Config{}
	              ^
e2e/pod.go:173:3: missing cases in switch of type v1.PodPhase: PodPending, PodUnknown (exhaustive)
		switch pod.Status.Phase {
		^
internal/journal/omap.go:74:2: return with no blank line before (nlreturn)
	return results, nil
	^
internal/journal/omap.go:69:3: return with no blank line before (nlreturn)
		return nil, err
		^
internal/journal/omap.go:67:4: return with no blank line before (nlreturn)
			return nil, util.JoinErrors(util.ErrKeyNotFound, err)
			^
internal/journal/voljournal.go:473:5: continue with no blank line before (nlreturn)
				continue
				^
internal/cephfs/volumemounter.go:126:4: break with no blank line before (nlreturn)
			break
			^
e2e/rbd.go:760:7: break with no blank line before (nlreturn)
						break
						^
internal/rbd/rbd_journal.go:411:4: break with no blank line before (nlreturn)
			break
			^
internal/rbd/rbd_util.go:468:4: continue with no blank line before (nlreturn)
			continue
			^
internal/util/util.go:137:4: continue with no blank line before (nlreturn)
			continue
			^
internal/util/vault.go:324:29: should rewrite http.NewRequestWithContext or add (*Request).WithContext (noctx)
	req, err := http.NewRequest(method, kms.VaultAddress+path, dataToSend)
	                           ^
INFO File cache stats: 98 entries of total size 818.8KiB 
INFO Memory: 394 samples, avg is 788.8MB, max is 1020.5MB 
INFO Execution took 44.962435553s                 
make: *** [Makefile:93: go-lint] Error 1

```

--- a/build.env
+++ b/build.env
@@ -20,8 +20,8 @@ GOLANG_VERSION=1.16.4
 GO111MODULE=on
 
 # static checks and linters
-GOLANGCI_VERSION=v1.27.0
-GOSEC_VERSION=v2.3.0
+GOLANGCI_VERSION=v1.39.0
+GOSEC_VERSION=v2.7.0
 
 # external snapshotter version
 # Refer: https://github.com/kubernetes-csi/external-snapshotter/releases

--- a/e2e/kms.go
+++ b/e2e/kms.go
@@ -9,7 +9,7 @@ import (
 )
 
 const (
-	// defaultVaultBackendPath is the default VAULT_BACKEND_PATH for secrets
+	// defaultVaultBackendPath is the default VAULT_BACKEND_PATH for secrets.
 	defaultVaultBackendPath = "secret/"
 )
 

--- a/e2e/staticpvc.go
+++ b/e2e/staticpvc.go
@@ -188,7 +188,7 @@ func validateRBDStaticPV(f *framework.Framework, appPath string, isBlock, checkI
 	return err
 }
 
-// nolint:gocyclo // reduce complexity
+// nolint:gocyclo,cyclop // reduce complexity
 func validateCephFsStaticPV(f *framework.Framework, appPath, scPath string) error {
 	opt := make(map[string]string)
 	var (

--- a/e2e/utils.go
+++ b/e2e/utils.go
@@ -527,7 +527,7 @@ func writeDataAndCalChecksum(app *v1.Pod, opt *metav1.ListOptions, f *framework.
 	return checkSum, nil
 }
 
-// nolint:gocyclo,gocognit,nestif // reduce complexity
+// nolint:gocyclo,gocognit,nestif,cyclop // reduce complexity
 func validatePVCClone(
 	totalCount int,
 	sourcePvcPath, sourceAppPath, clonePvcPath, clonePvcAppPath string,
@@ -728,7 +728,7 @@ func validatePVCClone(
 	validateRBDImageCount(f, 0, defaultRBDPool)
 }
 
-// nolint:gocyclo,gocognit,nestif // reduce complexity
+// nolint:gocyclo,gocognit,nestif,cyclop // reduce complexity
 func validatePVCSnapshot(
 	totalCount int,
 	pvcPath, appPath, snapshotPath, pvcClonePath, appClonePath string,

--- a/e2e/utils.go
+++ b/e2e/utils.go
@@ -1212,9 +1212,9 @@ func waitForJobCompletion(c kubernetes.Interface, ns, job string, timeout int) e
 type kubectlAction string
 
 const (
-	// kubectlCreate tells retryKubectlInput() to run "create"
+	// kubectlCreate tells retryKubectlInput() to run "create".
 	kubectlCreate = kubectlAction("create")
-	// kubectlDelete tells retryKubectlInput() to run "delete"
+	// kubectlDelete tells retryKubectlInput() to run "delete".
 	kubectlDelete = kubectlAction("delete")
 )
 

--- a/internal/cephfs/clone_test.go
+++ b/internal/cephfs/clone_test.go
@@ -23,6 +23,7 @@ import (
 )
 
 func TestCloneStateToError(t *testing.T) {
+	t.Parallel()
 	errorState := make(map[cephFSCloneState]error)
 	errorState[cephFSCloneComplete] = nil
 	errorState[cephFSCloneError] = ErrInvalidClone

--- a/internal/cephfs/controllerserver.go
+++ b/internal/cephfs/controllerserver.go
@@ -145,7 +145,7 @@ func checkContentSource(
 }
 
 // CreateVolume creates a reservation and the volume in backend, if it is not already present.
-// nolint:gocognit,gocyclo,nestif // TODO: reduce complexity
+// nolint:gocognit,gocyclo,nestif,cyclop // TODO: reduce complexity
 func (cs *ControllerServer) CreateVolume(
 	ctx context.Context,
 	req *csi.CreateVolumeRequest) (*csi.CreateVolumeResponse, error) {
@@ -508,7 +508,7 @@ func (cs *ControllerServer) ControllerExpandVolume(
 
 // CreateSnapshot creates the snapshot in backend and stores metadata
 // in store
-// nolint:gocyclo // golangci-lint did not catch this earlier, needs to get fixed late
+// nolint:gocyclo,cyclop // golangci-lint did not catch this earlier, needs to get fixed late
 func (cs *ControllerServer) CreateSnapshot(
 	ctx context.Context,
 	req *csi.CreateSnapshotRequest) (*csi.CreateSnapshotResponse, error) {

--- a/internal/cephfs/fsjournal.go
+++ b/internal/cephfs/fsjournal.go
@@ -55,7 +55,7 @@ because, the order of omap creation and deletion are inverse of each other, and 
 request name lock, and hence any stale omaps are leftovers from incomplete transactions and are
 hence safe to garbage collect.
 */
-// nolint:gocognit,gocyclo,nestif // TODO: reduce complexity
+// nolint:gocognit,gocyclo,nestif,cyclop // TODO: reduce complexity
 func checkVolExists(ctx context.Context,
 	volOptions,
 	parentVolOpt *volumeOptions,

--- a/internal/rbd/nodeserver.go
+++ b/internal/rbd/nodeserver.go
@@ -162,7 +162,7 @@ func healerStageTransaction(ctx context.Context, cr *util.Credentials, volOps *r
 //   - Create the staging file/directory under staging path
 //   - Stage the device (mount the device mapped for image)
 // TODO: make this function less complex.
-// nolint:gocyclo // reduce complexity
+// nolint:gocyclo,cyclop // reduce complexity
 func (ns *NodeServer) NodeStageVolume(
 	ctx context.Context,
 	req *csi.NodeStageVolumeRequest) (*csi.NodeStageVolumeResponse, error) {

--- a/internal/util/conn_pool_test.go
+++ b/internal/util/conn_pool_test.go
@@ -74,7 +74,7 @@ func (cp *ConnPool) fakeGet(monitors, user, keyfile string) (*rados.Conn, string
 	return conn, unique, nil
 }
 
-// nolint:paralleltest,tparallel
+// nolint:paralleltest // these tests cannot run in parallel
 func TestConnPool(t *testing.T) {
 	cp := NewConnPool(interval, expiry)
 	defer cp.Destroy()

--- a/internal/util/secretskms.go
+++ b/internal/util/secretskms.go
@@ -42,10 +42,10 @@ const (
 	kmsTypeSecretsMetadata = "metadata"
 
 	// metadataSecretNameKey contains the key which corresponds to the
-	// kubernetes secret name from where encryptionPassphrase is feteched
+	// kubernetes secret name from where encryptionPassphrase is feteched.
 	metadataSecretNameKey = "secretName"
 	// metadataSecretNamespaceKey contains the key which corresponds to the
-	// kubernetes secret namespace from where encryptionPassphrase is feteched
+	// kubernetes secret namespace from where encryptionPassphrase is feteched.
 	metadataSecretNamespaceKey = "secretNamespace"
 )
 

--- a/internal/util/vault.go
+++ b/internal/util/vault.go
@@ -113,7 +113,7 @@ func setConfigString(option *string, config map[string]interface{}, key string) 
 // these settings will be used when connecting to the Vault service with
 // vc.connectVault().
 //
-// nolint:gocyclo // iterating through many config options, not complex at all.
+// nolint:gocyclo,cyclop // iterating through many config options, not complex at all.
 func (vc *vaultConnection) initConnection(config map[string]interface{}) error {
 	vaultConfig := make(map[string]interface{})
 	keyContext := make(map[string]string)

--- a/internal/util/vault_sa_test.go
+++ b/internal/util/vault_sa_test.go
@@ -23,6 +23,7 @@ import (
 )
 
 func TestVaultTenantSAKMSRegistered(t *testing.T) {
+	t.Parallel()
 	_, ok := kmsManager.providers[kmsTypeVaultTenantSA]
 	assert.True(t, ok)
 }

--- a/internal/util/vault_tokens.go
+++ b/internal/util/vault_tokens.go
@@ -339,7 +339,7 @@ func (kms *VaultTokensKMS) setTokenName(config map[string]interface{}) error {
 // initCertificates updates the kms.vaultConfig with the options from config
 // it calls the kubernetes secrets and get the required data.
 
-// nolint:gocyclo // iterating through many config options, not complex at all.
+// nolint:gocyclo,cyclop // iterating through many config options, not complex at all.
 func (vtc *vaultTenantConnection) initCertificates(config map[string]interface{}) error {
 	vaultConfig := make(map[string]interface{})
 

--- a/scripts/golangci.yml.in
+++ b/scripts/golangci.yml.in
@@ -170,3 +170,5 @@ linters:
     # TODO: enable goerr113, see: https://github.com/ceph/ceph-csi/issues/1227
     - goerr113
     - forbidigo
+    # TODO: enable gomoddirectives
+    - gomoddirectives

--- a/scripts/golangci.yml.in
+++ b/scripts/golangci.yml.in
@@ -51,6 +51,8 @@ output:
 
 # all available settings of specific linters
 linters-settings:
+  cyclop:
+    max-complexity: 20
   errcheck:
     # report about not checking of errors in type assetions:
     #  `a := b.(MyStruct)`;

--- a/scripts/golangci.yml.in
+++ b/scripts/golangci.yml.in
@@ -169,3 +169,4 @@ linters:
     - exhaustivestruct
     # TODO: enable goerr113, see: https://github.com/ceph/ceph-csi/issues/1227
     - goerr113
+    - forbidigo

--- a/scripts/golangci.yml.in
+++ b/scripts/golangci.yml.in
@@ -166,5 +166,6 @@ linters:
     - wsl
     - funlen
     - testpackage
+    - exhaustivestruct
     # TODO: enable goerr113, see: https://github.com/ceph/ceph-csi/issues/1227
     - goerr113

--- a/scripts/golangci.yml.in
+++ b/scripts/golangci.yml.in
@@ -174,3 +174,7 @@ linters:
     - forbidigo
     # TODO: enable gomoddirectives
     - gomoddirectives
+    # TODO: enable gci
+    - gci
+    # TODO: enable wrapcheck linter, see: https://github.com/ceph/ceph-csi/pull/2268
+    - wrapcheck

